sequencing-server/src/defaultSeqBuilder.ts

```diff
@@ -42,3 +44,44 @@ export const defaultSeqBuilder: SeqBuilder = (
     steps: commands,
   });
 };
+
+// Order commands by time in expanded sequences.
+function sortCommandsByTime(commands: CommandStem<{} | []>[]): CommandStem<{} | []>[] {
+  const relativeTimeTracker: Record<string, Temporal.Instant> = {};
+  let previousAbsoluteTimestamp: Temporal.Instant;
+
+  for (const command of commands) {
+    // If the command is epoch-relative, complete, or the first command and relative then short circuit and don't try and sort.
+    if (
+      command.epochTime ||
+      (!command.absoluteTime && !command.epochTime && !command.relativeTime) ||
+      (command.relativeTime && commands.indexOf(command) === 0)
+    ) {
+      return commands;
+    }
+
+    // Keep track of the previously seen absolute time so we can convert the next relative timestamp we come across.
+    if (command.absoluteTime) {
```

Comment:
These kind of auto-cast checks aren't great because they have a lot of loopholes that are hard to keep track of (0 is falsy, empty string is falsy, etc etc). We should do explicit comparisons to undefined. The explicit checks should make the unsafe non-null assertions unnecessary as well.